src/utils/arkTransaction.ts

```diff
@@ -166,3 +168,159 @@ const nLocktimeMinSeconds = 500_000_000n;
 function isSeconds(locktime: bigint): boolean {
     return locktime >= nLocktimeMinSeconds;
 }
+
+/**
+ * Verify tapscript signatures on a transaction input
+ * @param tx Transaction to verify
+ * @param inputIndex Index of the input to verify
+ * @param requiredSigners List of required signer pubkeys (hex encoded)
+ * @param excludePubkeys List of pubkeys to exclude from verification (hex encoded, e.g., server key not yet signed)
+ * @param allowedSighashTypes List of allowed sighash types (defaults to [SigHash.DEFAULT])
+ * @throws Error if verification fails
+ */
+export function verifyTapscriptSignatures(
+    tx: Transaction,
+    inputIndex: number,
+    requiredSigners: string[],
+    excludePubkeys: string[] = [],
+    allowedSighashTypes: number[] = [SigHash.DEFAULT]
+): void {
+    const input = tx.getInput(inputIndex);
+
+    // Extract prevout information from witnessUtxo for the input being verified
+    if (!input.witnessUtxo) {
+        throw new Error(`Input ${inputIndex} is missing witnessUtxo`);
+    }
+
+    // Collect prevout scripts and amounts for ALL inputs (required for preimageWitnessV1)
+    const prevoutScripts: Uint8Array[] = [];
+    const prevoutAmounts: bigint[] = [];
+
+    for (let i = 0; i < tx.inputsLength; i++) {
+        const inp = tx.getInput(i);
+        if (!inp.witnessUtxo) {
+            throw new Error(`Input ${i} is missing witnessUtxo`);
+        }
+        prevoutScripts.push(inp.witnessUtxo.script);
+        prevoutAmounts.push(inp.witnessUtxo.amount);
+    }
+
+    // Verify tapScriptSig signatures
+    if (!input.tapScriptSig || input.tapScriptSig.length === 0) {
+        throw new Error(`Input ${inputIndex} is missing tapScriptSig`);
+    }
+
+    // Verify each signature in tapScriptSig
+    for (const [tapScriptSigData, signature] of input.tapScriptSig) {
+        const pubKey = tapScriptSigData.pubKey;
+        const pubKeyHex = hex.encode(pubKey);
+
+        // Skip verification for excluded pubkeys
+        if (excludePubkeys.includes(pubKeyHex)) {
+            continue;
+        }
+
+        // Extract sighash type from signature
+        // Schnorr signatures are 64 bytes, with optional 1-byte sighash appended
+        let sighashType = SigHash.DEFAULT;
+        let sig = signature;
+
+        if (signature.length === 65) {
+            // Last byte is the sighash type
+            sighashType = signature[64];
+            sig = signature.subarray(0, 64);
+        } else if (signature.length !== 64) {
+            throw new Error(
+                `Invalid signature length: ${signature.length} (expected 64 or 65)`
+            );
+        }
```

Comment:
```suggestion
       const sighashType = signature.length === 65 ? signature[64] : SigHash.DEFAULT;
       const sig = signature.subarray(0, 64);
```

we can remove the length check, `btc-signer` "Transaction" decode has robust sanity checks and verify the shnorr sig length